### PR TITLE
Added test for piral-cli npm

### PR DESCRIPTION
**Pull request by iotcloudarchitect**
# New Pull Request

For more information, see the `CONTRIBUTING` guide.

## Prerequisites

Please make sure you can check the following boxes:

- [ x] I have read the **CONTRIBUTING** document
- [ x] My code follows the code style of this project
- [x ] All new and existing tests passed

### Contribution Type

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [ ] Bug fix (non-breaking change which fixes an issue, please reference the issue id)
- [ ] New feature (non-breaking change which adds functionality, make sure to open an associated issue first)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] I have updated the documentation accordingly
- [x ] I have added tests to cover my changes

### Description

[Place a meaningful description here]

### Remarks

[Optionally place any follow-up comments, remarks, observations, or notes here for future reference]

### CODE REVIEW

File: src/tooling/piral-cli/src/common/npm.test.ts [hunk 1]
```diff
@@ -137,4 +174,202 @@ describe('NPM Module', () => {
     installPackage('npm', 'foo', '1.3', '.', '--a=b');
     expect(emitter).toHaveBeenCalledTimes(3);
   });
+
+  it('detectNpm finds package-lock.json', () => {
+    const emitter = jest.fn(() => ({ on: emitter, stdout: new Stream(), stdin: new Stream(), stderr: new Stream() }));
+    (cp as any).exec = emitter;
+    detectNpm('test')
+      .then(res => expect(res).toBeTruthy())
+      .catch(err => expect(err).toBeTruthy());
```
Comment (FlorianRappl) [inline src/tooling/piral-cli/src/common/npm.test.ts:183]: I don't understand this test. What does this do? We are also not properly awaiting it.

File: src/tooling/piral-cli/src/common/npm.test.ts [hunk 2]
```diff
@@ -137,4 +174,202 @@ describe('NPM Module', () => {
...
+  });
+
+  it('detectPnpm finds nppm-lock.yaml', () => {
+    const emitter = jest.fn(() => ({ on: emitter, stdout: new Stream(), stdin: new Stream(), stderr: new Stream() }));
+    (cp as any).exec = emitter;
+    detectPnpm('test')
```
Comment (FlorianRappl) [inline src/tooling/piral-cli/src/common/npm.test.ts:189]: Misses await. If it should throw please fall back to the pattern of evaluating the rejects.

File: src/tooling/piral-cli/src/common/npm.test.ts [hunk 3]
```diff
@@ -137,4 +174,202 @@ describe('NPM Module', () => {
...
+      .then(res => expect(res).toBeTruthy())
+      .catch(err => expect(err).toBeTruthy());
+  });
+
+  it('detectYarn finds yarn.lock', () => {
+    const emitter = jest.fn(() => ({ on: emitter, stdout: new Stream(), stdin: new Stream(), stderr: new Stream() }));
+    (cp as any).exec = emitter;
+    detectYarn('test')
```
Comment (FlorianRappl) [inline src/tooling/piral-cli/src/common/npm.test.ts:197]: Pretty much all tests here don't properly await. Skipping rest.

File: src/tooling/piral-cli/src/common/npm.test.ts (hunk 1 above)
Comment (iotcloudarchitect) [reply src/tooling/piral-cli/src/common/npm.test.ts]: Tests have been updated considering async now

File: src/tooling/piral-cli/src/common/npm.test.ts [hunk 4]
```diff
@@ -103,38 +150,182 @@ describe('NPM Module', () => {
...
+
+  it('detectYarn finds yarn.lock', async () => {
+    await detectYarn('test').then(result => expect(result).toBeTruthy());
+  });
+
+  it('uses npm to verify whether a particular package is included in monorepo package', async () => {
+    await isMonorepoPackageRef('npm', './').then(result => expect(result).toBeTruthy());
+  });
+
+  it('verfiies whether lerna config path is valid', async () => {
```
Comment (FlorianRappl) [inline src/tooling/piral-cli/src/common/npm.test.ts:189]: verifies

File: src/tooling/piral-cli/src/common/npm.test.ts [hunk 5]
```diff
@@ -103,38 +150,182 @@ describe('NPM Module', () => {
...
+  it('installs a package using the Pnpm command line tool without a version', async () => {
+    await installPackage('pnpm', 'foo').then(result => expect(result).toEqual(jsonValueString));
+  });
+
+  it('installs a package using the NPM command line tool with some flag', async () => {
+    await installPackage('npm', 'foo', '1.3', '.', '--a=b').then(result => expect(result).toEqual(jsonValueString));
+  });
+
+  it('detectNpm finds package-lock.json', async () => {
+    await detectNpm('test').then(result => expect(result).toBeTruthy());
```
[the remark this answers is not in the document]
Comment (iotcloudarchitect) [reply src/tooling/piral-cli/src/common/npm.test.ts:205]: Yes ;-)

**Pull request merged**